### PR TITLE
Fixes for Embeddinator after header changes

### PR DESCRIPTION
**Pull request by chamons**
- A number of our headers drifted into requiring Objc++ instead of Objc to compile
- Teach mmp/mtouch to link in C++ when handling E4K input

### CODE REVIEW

File: runtime/xamarin/runtime.h [hunk 1]
```diff
@@ -85,7 +85,7 @@ typedef struct __attribute__((packed)) {
 
 typedef struct __attribute__((packed)) {
 	const uint32_t *protocol_tokens; // an array of token references to managed interfaces that represent protocols
-	const Protocol **protocols; // the corresponding native protocols
+	const __unsafe_unretained Protocol **protocols; // the corresponding native protocols
```
Comment (chamons) [inline runtime/xamarin/runtime.h]: I am very uncertain if this is correct.

**Comment by spouliot** [comment]
Let's discuss if linking with c++ is really needed and, if not, what are the implications/other changes needed to avoid it.

File: runtime/xamarin/runtime.h (hunk 1 above)
Comment (rolfbjarne) [reply runtime/xamarin/runtime.h]: I believe this is correct, but please add a comment explaining why and when this is required (and is possible include the error message).

**Comment by spouliot** [comment]
`-lc++` situation cleared up. C++ is needed to compile (macios runtime) but **not** needed at runtime (so no linking required) -> removing do-not-merge label

**Comment by chamons** [comment]
https://github.com/xamarin/maccore/issues/655
https://github.com/xamarin/maccore/issues/656

**Pull request merged**